src/lib/journey/stages/generic.svelte

```diff
@@ -26,56 +32,69 @@
   export let step: WidgetStep;
   export let submitForm: () => void;
 
+  let alertNeedsFocus = false;
+  let callbackMetadataArray: CallbackMetadata[] = [];
+  let checkValidation: (callback: FRCallback) => boolean;
   let failureMessageKey = '';
-  let hasPrevError = false;
+  let stepMetadata: StepMetadata;
 
-  // TODO: Pull out and rework into a utility or helper
-  function checkValidation(callback: FRCallback) {
-    let failedPolices = callback.getOutputByName('failedPolicies', []);
-    if (failedPolices.length && !hasPrevError) {
-      hasPrevError = true;
-      return true;
+  function determineSubmission() {
+    // TODO: the below is more strict; all self-submitting cbs have to complete before submitting
+    // if (stepMetadata.isStepSelfSubmittable && isStepReadyToSubmit(callbackMetadataArray)) {
+
+    // The below variation is more liberal first self-submittable cb to call this wins.
+    if (stepMetadata.isStepSelfSubmittable) {
+      submitForm();
     }
-    return false;
   }
 
+  afterUpdate(() => {
+    alertNeedsFocus = !!failureMessage;
+  });
+
   $: {
+    checkValidation = initCheckValidation();
+    callbackMetadataArray = buildCallbackMetadata(step, checkValidation);
+    stepMetadata = buildStepMetadata(callbackMetadataArray);
```

Comment:
Here's where I'm calling for the build-up of the metadata, but I'm thinking about moving this into the journey's store.